### PR TITLE
Feature additions in pursuit of full comms with FE and microservices

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: src/witan/gateway/handler.clj [hunk 1]
```diff
@@ -40,21 +33,16 @@
...
-                         :event/created-at (iso-dt-now)
-                         :command/receipt receipt})
-      (catch Exception e
-        (println "Failed to dispatch event:" e)))))
+  [ch receipt event]
+  (try
+    (log/debug "Dispatching event to" ch)
+    (send-message! ch event)
+    (catch Exception e
+      (log/error "Failed to dispatch event:" e))))
```
Comment (otfrom) [inline src/witan/gateway/handler.clj]: is the correct behaviour here to just log and swallow the exception and return a nil?

File: src/witan/gateway/handler.clj [hunk 2]
```diff
@@ -74,13 +62,16 @@
 
 (defmethod handle-message
   :command
-  [ch {:keys [command/id command/key command/version] :as command} {:keys [kafka]}]
+  [ch {:keys [command/id command/key command/version] :as command} {:keys [kafka connections]}]
   (let [receipt (java.util.UUID/randomUUID)
         now (iso-dt-now)]
-    (swap! receipts assoc receipt {:ch ch :at (t/now)})
+    (p/add-receipt! connections (partial dispatch-event! ch receipt) receipt)
```
Comment (otfrom) [inline src/witan/gateway/handler.clj]: what should we do if this fails?

**Comment by otfrom** [comment]
There are some error handling things that we probably want to think about here, but :shipit:

File: src/witan/gateway/handler.clj (hunk 1 above)
Comment (acron0) [inline src/witan/gateway/handler.clj]: It's the intended behaviour at the moment, yes. If 'send-message!' has thrown an exception then, for whatever reason, we're unable to put anything on Kafka

File: src/witan/gateway/handler.clj (hunk 2 above)
Comment (acron0) [inline src/witan/gateway/handler.clj]: Your question is a bit vague. Do you mean literally this line? Or the whole `handle-message` fn?

**Pull request merged**